userguide/cpp.md

```diff
@@ -0,0 +1,198 @@
+
+##### Setup
+
+Using the C++ API requires building Turi Create from source, starting with a clone of the [repository](git@github.com:apple/turicreate.git). You should consider checking out a particular version tag (for instance `v4.1`) in order to keep API stability as you proceed (the API may change in master). Before proceeding to use the C++ API, run `./configure` to set up the build system for the project, and run `make -j8` from within `release/src/unity` to build the library artifacts that your application will depend on.
+
+Currently, the C++ API is not packaged for easy consumption outside of the Turi Create Python package. As a result, some care is needed when compiling and linking against Turi Create. This may cause compatibility issues since those compilation and linking flags may not be compatible with flags used elsewhere. Required compilation flags for your project include:
+
+* `-std=c++11` - some headers exposed by Turi Create require the use of C++11 or later.
+* `-stdlib=libc++` - may be needed on macOS (probably not on Linux).
+* Assuming PATH_TO_TC refers to the location where you cloned the Turi Create repository, the following directories must be added to the include paths with `-I`, and to the link paths with `-L`:
+  * `-IPATH_TO_TC/turicreate/src`
+  * `-IPATH_TO_TC/turicreate/deps/local/include`
+  * `-IPATH_TO_TC/turicreate/src/external/armadillo/include`
+  * `-LPATH_TO_TC/turicreate/release/src/unity`
+* Depending on your compiler toolchain, the following defines may be needed (note: these are autodetected for Turi Create itself with `./configure`, but the values must be passed on to your own build process:
+* `-DHASH_FOR_UINT128_DEFINED` - set this if your compiler toolchain supports `unsigned __int128`.
+* `-DHASH_FOR_INT128_DEFINED` - set this if your compiler toolchain supports `__int128`.
+* `-lunity_shared` to link against `libunity_shared.so`.
+* `PATH_TO_TC/turicreate/release/src/unity/libunity.a` to statically link libunity.a into your project. Note that this is needed in addition to the dynamic library listed above.
+
+##### Example: make predictions in C++ from a model trained in Python
+
+Let's assume we're starting with the model that results from the introductory example in the [boosted trees classifier](https://github.com/apple/turicreate/blob/3490286b27ff5d79cb90d09fe026d5671ce990c7/userguide/supervised-learning/boosted_trees_classifier.md#gradient-boosted-regression-trees) section of the user guide. The model is available as [mushroom.tcmodel.zip](https://github.com/apple/turicreate/files/1654991/mushroom.tcmodel.zip) as well.
+
+To make predictions from this model in Python, you could run:
+```python
+import turicreate as tc
+data =  tc.SFrame.read_csv('turicreate/lfs/datasets/xgboost/mushroom.csv')
+model = tc.load_model('temp4/mushroom.tcmodel')
```

Comment:
The file path should probably just be 'mushroom.tcmodel' rather than 'temp4/mushroom.tcmodel'.